ament_lint_cmake/ament_lint_cmake/cmakelint.py

```diff
@@ -382,6 +386,16 @@ def IsValidFile(filename):
     return filename.endswith('.cmake') or os.path.basename(filename).lower() == 'cmakelists.txt'
 
 def ProcessFile(filename):
+    # Store and then restore the filters to prevent pragmas in the file from persisting.
+    original_filters = list(_lint_state.filters)
+    try:
+        return _ProcessFile(filename)
```

Comment:
I don't think so, because the `_lint_state` object is a global and `_ProcessFile` (potentially) mutates it. If I were to implement this in the `_ProcessFile` function then I would just wrap the whole thing in a try/finally (which is what I've done). Is there something you were thinking I should do instead?